rubber_duck.py

```diff
@@ -5,8 +5,13 @@
 import traceback as tb
 from typing import TypedDict, Protocol, ContextManager
 from openai import AsyncOpenAI
+import openai
 from openai.types.chat.chat_completion import ChatCompletion
 from quest import step, queue
+import httpx
+
+from unittest.mock import Mock
```

Comment:
Let's stick to `pytest` (don't use `unittest`). 

Also, importing testing resources in production modules is almost always an indication of something fishy going on. Do you actually use `Mock` in this file?